From: SZALAY Attila <sasa@debian.org>
Date: Fri, 20 Jan 2012 13:29:40 +0100
Subject: compile_problems2

===================================================================
---
 src/memtrace.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/memtrace.cc b/src/memtrace.cc
index f30f066..fcb5533 100644
--- a/src/memtrace.cc
+++ b/src/memtrace.cc
@@ -688,7 +688,7 @@ z_malloc(size_t size, gpointer backt[])
       if (temp_brk > MEMTRACE_TEMP_HEAP_SIZE)
         {
           temp_brk = 0;
-          assert(0);
+          assert(backtrace_info == 0);
         }
     }
   else
