From: SZALAY Attila <sasa@debian.org>
Date: Fri, 20 Jan 2012 13:29:40 +0100
Subject: compile_problems2

===================================================================
---
 src/memtrace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/memtrace.c b/src/memtrace.c
index eda45de..3ee4ef7 100644
--- a/src/memtrace.c
+++ b/src/memtrace.c
@@ -693,7 +693,7 @@ z_malloc(size_t size, gpointer backt[])
         {
           backtrace_info = temp_brk;
           temp_brk = 0;
-          assert(0);
+          assert(backtrace_info == 0);
         }
     }
   else
