From: SZALAY Attila <sasa@debian.org>
Date: Fri, 20 Jan 2012 13:29:40 +0100
Subject: compile_problems3

---
 src/zorp/io.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/zorp/io.h b/src/zorp/io.h
index 264a085..8152388 100644
--- a/src/zorp/io.h
+++ b/src/zorp/io.h
@@ -35,9 +35,9 @@ void z_fd_set_our_tos(gint fd, guint8 tos);
 
 #else
 
-#define z_fd_get_peer_tos(fd, tos)
-#define z_fd_get_our_tos(fd, tos)
-#define z_fd_set_our_tos(fd, tos)
+#define z_fd_get_peer_tos(fd, tos) {}
+#define z_fd_get_our_tos(fd, tos) {}
+#define z_fd_set_our_tos(fd, tos) {}
 
 #endif
 
