From: SZALAY Attila <sasa@debian.org>
Date: Sun, 28 Jun 2015 21:25:48 +0200
Subject: Removed Compile-date field because it prevent reliable rebuilding

---
 src/misc.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/misc.cc b/src/misc.cc
index 6ac746b..f96ca72 100644
--- a/src/misc.cc
+++ b/src/misc.cc
@@ -563,14 +563,13 @@ z_libzorpll_version_info(void)
   g_snprintf(buf, sizeof(buf),
              "libzorpll %s\n"
              "Revision: %s\n"
-             "Compile-Date: %s %s\n"
              "Trace: %s\n"
              "MemTrace: %s\n"
              "Caps: %s\n"
              "Debug: %s\n"
              "StackDump: %s\n",
 
-             ZORPLIBLL_VERSION, ZORPLIBLL_REVISION, __DATE__, __TIME__,
+             ZORPLIBLL_VERSION, ZORPLIBLL_REVISION,
              ON_OFF_STR(ZORPLIB_ENABLE_TRACE),
              ON_OFF_STR(ZORPLIB_ENABLE_MEM_TRACE),
              ON_OFF_STR(ZORPLIB_ENABLE_CAPS),
